samples/android/app/src/androidTest/java/org/mozilla/samples/gleancore/pings/DeletionRequestPingTest.kt

```diff
@@ -0,0 +1,147 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.samples.gleancore.pings
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.closeSoftKeyboard
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import org.junit.Assert.assertEquals
+import org.mozilla.samples.gleancore.R
+
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.samples.gleancore.MainActivity
+import org.mozilla.samples.gleancore.getPingServer
+import androidx.test.uiautomator.UiDevice
+import androidx.work.WorkInfo
+import androidx.work.WorkManager
+import androidx.work.testing.WorkManagerTestInitHelper
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import mozilla.telemetry.glean.testing.GleanTestLocalServer
+import org.json.JSONObject
+import org.junit.Assert.assertNotEquals
+import org.junit.Before
+import org.mozilla.samples.gleancore.getPingServerPort
+import java.util.concurrent.TimeUnit
+
+@RunWith(AndroidJUnit4::class)
+class DeletionRequestPingTest {
+    @get:Rule
+    val activityRule: ActivityTestRule<MainActivity> = ActivityTestRule(MainActivity::class.java)
+
+    @get:Rule
+    val gleanRule = GleanTestLocalServer(getPingServerPort())
+
+    private val context: Context
+        get() = ApplicationProvider.getApplicationContext()
+
+    @Before
+    fun clearWorkManager() {
+        WorkManagerTestInitHelper.initializeTestWorkManager(context)
+    }
+
+    private fun waitForPingContent(
+        pingName: String,
+        maxAttempts: Int = 3
+    ): JSONObject?
+    {
+        val server = getPingServer()
+
+        var attempts = 0
```

Comment:
Wow, why do we need this? Are we failing because we are not receiving the pings?

Reply:
We should not, we're already waiting 20secs to receive something, let me simplify that